src/Avalonia.Diagnostics/Diagnostics/Views/PropertyValueEditorView.cs

```diff
@@ -0,0 +1,394 @@
+﻿using System;
+using System.ComponentModel;
+using System.Globalization;
+using System.Reflection;
+using Avalonia.Controls;
+using Avalonia.Controls.Primitives;
+using Avalonia.Controls.Shapes;
+using Avalonia.Data;
+using Avalonia.Data.Converters;
+using Avalonia.Diagnostics.Controls;
+using Avalonia.Diagnostics.ViewModels;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.Layout;
+using Avalonia.Markup.Xaml.Converters;
+using Avalonia.Media;
+using Avalonia.Reactive;
+
+namespace Avalonia.Diagnostics.Views
+{
+    internal class PropertyValueEditorView : UserControl
+    {
+        private static readonly Geometry ImageIcon = Geometry.Parse(
+            "M12.25 6C8.79822 6 6 8.79822 6 12.25V35.75C6 37.1059 6.43174 38.3609 7.16525 39.3851L21.5252 25.0251C22.8921 23.6583 25.1081 23.6583 26.475 25.0251L40.8348 39.385C41.5683 38.3608 42 37.1058 42 35.75V12.25C42 8.79822 39.2018 6 35.75 6H12.25ZM34.5 17.5C34.5 19.7091 32.7091 21.5 30.5 21.5C28.2909 21.5 26.5 19.7091 26.5 17.5C26.5 15.2909 28.2909 13.5 30.5 13.5C32.7091 13.5 34.5 15.2909 34.5 17.5ZM39.0024 41.0881L24.7072 26.7929C24.3167 26.4024 23.6835 26.4024 23.293 26.7929L8.99769 41.0882C9.94516 41.6667 11.0587 42 12.25 42H35.75C36.9414 42 38.0549 41.6666 39.0024 41.0881Z");
+
+        private static readonly Geometry GeometryIcon = Geometry.Parse(
+            "M23.25 15.5H30.8529C29.8865 8.99258 24.2763 4 17.5 4C10.0442 4 4 10.0442 4 17.5C4 24.2763 8.99258 29.8865 15.5 30.8529V23.25C15.5 18.9698 18.9698 15.5 23.25 15.5ZM23.25 18C20.3505 18 18 20.3505 18 23.25V38.75C18 41.6495 20.3505 44 23.25 44H38.75C41.6495 44 44 41.6495 44 38.75V23.25C44 20.3505 41.6495 18 38.75 18H23.25Z");
+
+        private static readonly ColorToBrushConverter Color2Brush = new();
+
+        private readonly CompositeDisposable _cleanup = new();
+        private PropertyViewModel? Property => (PropertyViewModel?)DataContext;
+
+        protected override void OnDataContextChanged(EventArgs e)
+        {
+            base.OnDataContextChanged(e);
+
+            Content = UpdateControl();
+        }
+
+        protected override void OnDetachedFromVisualTree(VisualTreeAttachmentEventArgs e)
+        {
+            base.OnDetachedFromVisualTree(e);
+
+            _cleanup.Clear();
+        }
+
+        private static bool ImplementsInterface<TInterface>(Type type)
+        {
+            var interfaceType = typeof(TInterface);
+            return type == interfaceType || type.GetInterface(interfaceType.FullName!) != null;
```

Comment:
Why not `typeof(TInterface).IsAssignableFrom(type)`? Same with geometry below.

Reply:
Because `typeof(object).IsAssignableFrom()` or `typeof(IAvaloniaObject).IsAssignableFrom()` (or any base interface) also returns true. I was using it a the beginning but it was causing too many edge cases.

Reply:
AFAIK, `GetInterface` would find any implemented interface, including those implemented indirectly (Roslyn marks them as implemented directly: https://sharplab.io/#v2:CYLg1APgAgTAjAWAFDIJYDsAuBTATgMwEMBjbAAgElEkBvZAX2TSzyNMpjJEurqUZRIoAZjKwyAYS4dkfAUA).

Or do you mean something different when you say "base interface"?

Reply:
Went back to `typeof(TInterface).IsAssignableFrom(type)`. I got it backwards the first time ... :D